### PR TITLE
Feature: Crowdsourcing Forms

### LINKED ISSUE #113: Crowdsourcing tap information
Ability for in-app native crowdsourcing of PHLASK tap data

PHLASK tap data crowdsourcing would fall into two categories:
1. New PHLASK taps (food & water)
2. Existing PHLASK tap information

**New PHLASK Taps:** Users should be able to submit new taps and any corresponding information they have for a tap. 
**Existing PHLASK Taps:** Users can submit information that is missing from a tap OR propose changes to existing information

Crowdsourced Information:
Users should be able to submit crowdsourced information for any PHLASK Tap's data fields

Verification Process
Crowdsourced information is submitted to a separate database and manually reviewed before being approved and merged into the primary PHLASK Tap database

### PR DESCRIPTION
**Pull request by nwyin**
Hello! I am so excited to finally get this PR in. Super sorry it took ~2 months to develop everything (and there's still more we could do), but now I feel like there have been enough substantial changes and minimum functionality that it makes sense to get this reviewed and tested by more people than just myself.

I've tried my best to break down and summarize all the work that's been done on this branch (along with my best attempt at good commit messages), but I know that it is confusing to look at other people's work. Please reach out to me at @tbn33 on Slack if there are any questions or concerns.

## Summary of Changes
**TL;DR: Crowdsourcing forms are up and functional, and are ready for testing on the `test` environment**

-  `AddTapModal.js` has been changed to `AddResouceModal.js`, and all related components, styles, and utility logic are grouped together in `src/components/AddResourceModal`
    - It's my recommendation that as components grow more complex we move to this structure for new and existing components to make it easier for new (and existing) developers to identify the different domains and functions of the app
- `AddTapModal.js` has additional state and event handlers to accommodate new resource types. Specifically, the "view" of the modal is controlled by a new state attribute called `formStep`, which is modified to show the different forms for different resources.
- To try and reduce copy + pasted code, resource forms have been split into their own sub-components, which are composed of the `ShareFormFields.jsx` component and then additional custom fields specific to that resource
- Unrelated to the crowdsourcing form, by the `cypress.json` file has been modified such that none of the example tests are run. I've made this change as the example tests are non-relevant to the Phlask repo and make running tests take more time. 
- I've created additional test databases on Firebase for foraging and bathroom resources, title `phlask-web-map-test-foraging-verify` and `phlask-web-map-test-bathroom-verify` respectively.
- Address suggestion/autocomplete on each crowdsourcing form is functional. This relates to issue #196.

## Known Issues + Areas of Improvement
- `firebase` is subject to throwing errors as a result of having multiple database connections open at the same time. Specifically, when using the React Dev Tools add-on on Chrome, an error is thrown when using the `Profiler` tab to try and read state. I cannot reproduce the bug otherwise, so for the time being I am assuming this bug is caused by some weird developer interaction.
- **beta and prod databases have not been created on the Firebase console yet**
    - additionally, we should double-check that **read and write access to these databases are set to true** for testing, otherwise, an error will be thrown and cause a crash  
  - Unit and integration tests need to be written so we can automatically verify that forms are working as intended, data is being uploading to Firebase correctly, and that we can then read data from an API request to the database. Issue #199 is specifically for this issue
  - I have not extensively tested the address autocomplete functionality, and there may be edge cases or issues related to network/latency I'm not aware of.
  - **all state of the form is uploaded to every resource type**. this means that each resource type has information that's not relevant to that resource type (e.g. `foraging_food_type` field existing on a `water` entry). This was for convenience, as client-side code is not affected by these extra fields existing on the resource entry, **but I recognize this might lead to latency issues as there's more data per entry on each resource**
      - one way to address this is writing additional logic to check what resource type we are submitting, and constructing the appropriate state object for that resource type, and submitting that (as opposed to just submitting the whole form state). This can be found in the `onSubmit` function in `AddResourceModal.js`

### CODE REVIEW

**Review by gcardonag — APPROVED** [review]

**Review by affabillyty — APPROVED** [review]
Wahoo! 🚀
